### ISSUE TITLE
prevent-construct-id-collision: false positive when the scope is created inside the loop

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/awscdk-lint#569]
### Describe the bug

`prevent-construct-id-collision` looks only at the construct ID (2nd argument) and at whether
the `new` expression is lexically inside a loop. It ignores the scope (1st argument).

Construct IDs only have to be unique **within a scope** — this is stated in the rule's own
documentation ("Since Construct IDs must be unique within a specific scope, this causes
deployment errors"). When the scope itself is created fresh on every iteration, a literal ID
cannot collide, yet the rule reports it.

**Reproduction**

`src/fixture.ts`:

```ts
import { App, Stack } from "aws-cdk-lib";
import { Bucket } from "aws-cdk-lib/aws-s3";

const app = new App();

for (const stage of ["dev", "prod"]) {
  const stack = new Stack(app, `${stage}Stack`);
  new Bucket(stack, "Bucket");
}
```

`eslint.config.mjs`:

```js
import cdkPlugin from "eslint-plugin-awscdk";
import tseslint from "typescript-eslint";

export default [
  {
    files: ["**/*.ts"],
    plugins: { awscdk: cdkPlugin },
    languageOptions: {
      parser: tseslint.parser,
      parserOptions: { projectService: true, tsconfigRootDir: import.meta.dirname },
    },
    rules: { "awscdk/prevent-construct-id-collision": "error" },
  },
];
```

`.oxlintrc.json`:

```json
{
  "jsPlugins": ["oxlint-plugin-awscdk"],
  "options": { "typeAware": true },
  "rules": { "awscdk/prevent-construct-id-collision": "error" }
}
```

Commands and output:

```
$ npx eslint --config eslint.config.mjs src/fixture.ts

/path/to/repro/src/fixture.ts
  8:21  error  Construct ID 'Bucket' is a literal value inside a loop. This may cause ID collisions. Use a variable that changes per iteration instead  awscdk/prevent-construct-id-collision

✖ 1 problem (1 error, 0 warnings)

$ npx oxlint --format=unix -c .oxlintrc.json src/fixture.ts
src/fixture.ts:8:21: Construct ID 'Bucket' is a literal value inside a loop. This may cause ID collisions. Use a variable that changes per iteration instead. [Error/awscdk(prevent-construct-id-collision)]

1 problem
```

The reported code synthesizes without any collision — each iteration creates its own `Stack`
scope (`synth-check.cjs`, run with the same `aws-cdk-lib` 2.266.0):

```js
const { App, Stack } = require("aws-cdk-lib");
const { Bucket } = require("aws-cdk-lib/aws-s3");

const app = new App();
for (const stage of ["dev", "prod"]) {
  const stack = new Stack(app, `${stage}Stack`);
  new Bucket(stack, "Bucket");
}
const asm = app.synth();
for (const s of asm.stacks) {
  console.log(s.stackName, "->", Object.keys(s.template.Resources).join(", "));
}
```

```
$ node synth-check.cjs
devStack -> Bucket83908E77
prodStack -> Bucket83908E77
```

For contrast, an actual collision (two constructs with the same ID in the *same* scope) throws:

```
ERROR: There is already a Construct with name 'MyBucket' in Stack2 [S]
```

The same false positive occurs with an iteration-method callback:

```ts
names.forEach((name) => {
  const group = new Group(this, `${name}Group`);
  new Bucket(group, "Bucket"); // reported, but `group` differs per iteration
});
```

### Expected Behavior

No diagnostic when the scope passed as the first argument is a binding that is (re)created
inside the same loop body — there is no possible ID collision in that case.

### Current Behavior

The literal ID is reported regardless of the scope, so the recommended fix (adding a per-item
suffix to the ID) has to be applied to code that is already correct, and it changes the
CloudFormation logical IDs of resources for no reason.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.1, oxlint-plugin-awscdk 1.0.1

### Language Version

TypeScript 6.0.3 (eslint-plugin-awscdk) / TypeScript 7.0.2 via oxlint-tsgolint (oxlint-plugin-awscdk)

### Additional Context

```yaml
# machine-readable summary
rules: ["prevent-construct-id-collision"]
symptom: ["false-positive"]
affected_packages: ["eslint-plugin-awscdk", "oxlint-plugin-awscdk"]
root_cause: "packages/*/src/rules/prevent-construct-id-collision.ts:48"
verified_on: "main @ 6044b32 (eslint-plugin-awscdk 4.4.1, oxlint-plugin-awscdk 1.0.1)"
related_issues: []
```

Root cause: `validateConstructIdInLoop()` / `isInsideLoop()` only consider `node.arguments[1]`
and the ancestor chain; `node.arguments[0]` is never examined.

- `packages/eslint/src/rules/prevent-construct-id-collision.ts:48`
- `packages/oxlint/src/rules/prevent-construct-id-collision.ts:51`

A minimal improvement would be to skip the report when the first argument is an `Identifier`
whose declaration is a `VariableDeclarator` inside the same loop body / iteration-callback body.

Reproduction environment:

- macOS (darwin 23.1.0), Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.266.0 / constructs 10.8.1
- ESLint 10.8.1 + typescript-eslint 8.67.0 + TypeScript 6.0.3
- oxlint 1.79.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.12.2)
